pytest.ini

```diff
@@ -1,3 +1,57 @@
 [pytest]
 addopts = -rs --ignore=setup.py --ignore=test_requirements.txt
 norecursedirs = .git build dist tmp* .eggs
+text_file_format = rst
+doctest_plus = enabled
+doctest_optionflags =
+    NORMALIZE_WHITESPACE
+    ELLIPSIS
+    FLOAT_CMP
+doctest_norecursedirs =
+    extern
+    sherpa-test-data
+    # Blacklist everything and then remove from this list when it's ready
+    docs/_examples
+    docs/code
+    docs/data
+    docs/developer
+    docs/examples
+    docs/extra
+    docs/fit
+    docs/install.rst
+    docs/mcmc
+    docs/model_classes
+    docs/models
+    docs/optimisers
+    docs/overview
+    docs/plots
+    docs/statistics
+    docs/ui
+    sherpa/__init__.py
+    sherpa/astro
+    sherpa/conftest.py
+    sherpa/estmethods
+    sherpa/fit.py
+    sherpa/image
+    sherpa/include
+    sherpa/io.py
+    sherpa/models
+    sherpa/optmethods
+    sherpa/plot
+    sherpa/sim
+    sherpa/static
+    sherpa/stats
+    sherpa/tests
+    sherpa/ui
+    sherpa/utils
+
+doctest_subpackage_requires =
+    sherpa/astro/io/crates_backend.py = pycrates
+    sherpa/astro/io/pyfits_backend.py = astropy
+    sherpa/plot/pylab_backend.py = matplotlib
+    # Don't know how to check if ds9 and xspec are installed for this purpose.
+    # They are not Python packages, but the following syntax will skip these files
+    # since Python packages named ds9 or xspec don't exist and thus won't be found.
+    sherpa/image/*ds9_backend.py = ds9
+    sherpa/image/DS9.py = ds9
+    scripts/check_xspec_update.py = xspec
```

Comment:
Three thoughts:


1. So does this mean that at the moment we will never run tests of these particular files?
2. is the `*` in the `ds9_backend` line intended or just a typo? I'd expect an explicit path here, so I would assume a type
3. I also think that we probably want to just always ignore the `scripts` directory (as noted above) and so we could drop the last line

Reply:
1. We will never run the doctests in the files that have a dependence on "xspec" or "ds9" since we don't have python packages with those names. However, I've since learned a better way to accomplish this task (define a pytest fixtures that tests if xspec or ds9 are available and call `pytest.skip("ds9 not available")` when it's not. However, those files are not very high on my priority list.
2. It's intentional to cover `ds9_backend.py` and `pyds9_backend.py`, but not `dummy_backend.py`. I could spell it out, but again this is probably going to be reworked when we get to those files anyway.
3. Ok.